beam/www/beam/components/ControlButtons.vue

```diff
@@ -0,0 +1,65 @@
+<template>
+	<div class="control-buttons">
+		<button class="btn" @click="onAmend" v-if="docstatus === 2">Amend</button>
+		<template v-else>
+			<button class="btn" @click="onSave" :disabled="!(docstatus === 0 && doctypeName === '')">Save</button>
+			<button class="btn" @click="onSubmit" :disabled="!(docstatus === 0 && doctypeName !== '')">Submit</button>
+			<button class="btn" @click="onCancel" :disabled="!(docstatus === 1)">Cancel</button>
+		</template>
+	</div>
+</template>
+
+<script setup lang="ts" generic="T extends ParentDoctype">
+import { ParentDoctype } from '@/types';
+import { defineProps, ref } from 'vue'
+const docstatus = ref(0)
+const doctypeName = ref("")
+
+const props = defineProps<{
+	onCreate: () => Promise<{ data: T }>
```

Comment:
Add exception and response to the types too.

Optionally, this could be simplified into a type as well. Something like:

```ts
export type DocActionResponse<T> = {
	data: T
	exception: string
	response: Response
}
```

Then the prop types can just be:

```ts
onCreate: () => Promise<DocActionResponse<T>>
```